winfsp.go

```diff
@@ -46,10 +49,51 @@ func (mfs *FS) MountWinFSP(mntPoint string) (MountRunner, error) {
 	return s, nil
 }
 
+// foFSImpl is the interface implemented by the
+// the "base" filesystem returned by [gofs.New].
+type goFSImpl interface {
+	winfsp.BehaviourBase
+	winfsp.BehaviourGetSecurityByName
+	winfsp.BehaviourCreate
+	winfsp.BehaviourOverwrite
+	winfsp.BehaviourReadDirectory
+	winfsp.BehaviourGetFileInfo
+	winfsp.BehaviourGetSecurity
+	winfsp.BehaviourGetVolumeInfo
+	winfsp.BehaviourSetVolumeLabel
+	winfsp.BehaviourSetBasicInfo
+	winfsp.BehaviourSetFileSize
+	winfsp.BehaviourRead
+	winfsp.BehaviourWrite
+	winfsp.BehaviourFlush
+	winfsp.BehaviourCanDelete
+	winfsp.BehaviourCleanup
+	winfsp.BehaviourRename
+}
+
+// fspBaseBehavior implements [winfsp.BehaviourBase] by
+// wrapping a gofs.New filesystem.
+//
+// It exists purely so we can unset the windows.FILE_OPEN_NO_RECALL bit
+// from OpenFile calls. Otherwise PowerShell sets it and then
+// winfsp/gofs rejects the open, which is arguably a bug.
```

Comment:
I have to admit I am not 100% following this sequence of events. Is PowerShell involved because we're running a build from PowerShell, and the issue is you can't have 2 different processes open the same file with this bit set?